hook-bootkit/main.go

```diff
@@ -144,16 +147,35 @@ func main() {
 
 	fmt.Printf("Pulling image [%s]", imageName)
 
-	out, err := cli.ImagePull(ctx, imageName, pullOpts)
+	// TODO: Ideally if this function becomes a loop that runs forever and keeps retrying
+	// anything that failed, this retry would not be needed. For now, this addresses the specific
+	// race condition case of when the linuxkit network or dns is in the process of, but not quite
+	// fully set up yet.
+
+	var out io.ReadCloser
+	imagePullOperation := func() error {
+		out, err = cli.ImagePull(ctx, imageName, pullOpts)
+		if err != nil {
+			fmt.Printf("Image pull failure %s, %v\n", imageName, err)
+		}
+		return err
```

Comment:
This is a bit unclear what value of err is returned. I feel it preferable to return the error in the `err != nil` block and return a final `nil`. -- [ref](https://go.dev/doc/effective_go#if:~:text=The%20code%20reads%20well%20if%20the%20successful%20flow%20of%20control%20runs%20down%20the%20page%2C%20eliminating%20error%20cases%20as%20they%20arise)

```go
if err != nil {
	return errors.New(fmt.Sprintf("Image pull failure %s, err: %w", imageName, err))
}
return nil
```